libraries/architecture/src/main/kotlin/io/element/android/libraries/architecture/ParentNodeExt.kt

```diff
@@ -48,3 +49,23 @@ suspend inline fun <reified N : Node, NavTarget : Any> ParentNode<NavTarget>.wai
             continuation.cancel()
         }
     }
+
+/**
+ * Wait for a child to be attached to the parent node, only using the NavTarget
+ */
+suspend inline fun <NavTarget : Any> ParentNode<NavTarget>.waitForNavTargetAttached(crossinline predicate: (NavTarget) -> Boolean) =
+    suspendCancellableCoroutine { continuation ->
+        lifecycleScope.launch {
+            children.collect { childMap ->
+                val node = childMap.entries
+                    .map { it.key.navTarget }
+                    .lastOrNull(predicate)
+                if (node != null && !continuation.isCompleted) {
+                    continuation.resume(Unit)
+                    cancel()
```

Comment:
I think `cancel()` should also be invoked after line 45, since once the continuation is resumed, there is no need to invoke the predicated until the Node is destroyed.

But I do not want to do it in that PR.